operator/controllers/syncer/csi_syncer.go

```diff
@@ -416,7 +415,7 @@ func (s *csiControllerSyncer) ensureAttacherPodSpec(secrets []corev1.LocalObject
 		ServiceAccountName: config.GetNameForResource(config.CSIAttacherServiceAccount, s.driver.Name),
 		ImagePullSecrets:   secrets,
 		PriorityClassName:  "system-node-critical",
-		//SecurityContext:    ensurePodSecurityContext(config.RunAsUser, config.RunAsGroup, true),
+		SecurityContext:    ensurePodSecurityContext(config.RunAsUser, config.RunAsGroup, false),
```

Comment:
SecurityContext is now set at 2 places pod and container, with a common field RunAsNonRoot set twice.
Instead of setting defaults, setting an empty struct like below would work here?
```
podSecurityContext := corev1.PodSecurityContext{}
SecurityContext: &podSecurityContext
```

Reply:
Empty struct override would not work here that's why we went for setting default values fix. Setting at 2 places should not be a problem as ideally the fix is required only at container level but setting default values at pod level would be a temporary fix which might not be required if we come up with proper solution with mergo or alternative to same.